File: backend/app/schemas/lead.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, EmailStr


class LeadCreate(BaseModel):
    email: EmailStr
    source: str


class LeadRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    email: EmailStr
    source: str
    created_at: datetime
    ranking_score: float | None = None
    ranking_explanation: str | None = None
    last_ranked_at: datetime | None = None
    assigned_to: str | None = None
    routing_reason: str | None = None
    last_routed_at: datetime | None = None

File: frontend/src/types/lead.ts
/** Mirrors backend ``LeadRead`` JSON (snake_case). */
export type Lead = {
  id: string;
  email: string;
  source: string;
  created_at: string;
  full_name?: string | null;
  company?: string | null;
  status?: string | null;
  updated_at?: string | null;
  ranking_score: number | null;
  ranking_explanation: string | null;
  last_ranked_at: string | null;
  assigned_to: string | null;
  routing_reason: string | null;
  last_routed_at: string | null;
};
